File: modules/spring-core/testdata/java/providers/linemarkers/beans/autowired/FooFormatter.java
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component("fooFormatter")
class FooFormatter {
    public String format() {
        return "foo";
    }
}

@Component
class FooServiceField {
    @Autowired
    private FooFormatter fooFormatter;
}

@Component
class FooServiceSetter {
    private FooFormatter fooFormatter;
    @Autowired
    public void setFormatter(FooFormatter fooFormatter) {
        this.fooFormatter = fooFormatter;
    }
}

@Component
class FooServiceConstructor {
    private FooFormatter fooFormatter;
    @Autowired
    public FooServiceConstructor(FooFormatter fooFormatter) {
        this.fooFormatter = fooFormatter;
    }
}

File: modules/spring-core/testdata/kotlin/providers/linemarkers/beans/autowired/FooFormatter.kt
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component("fooFormatter")
internal class FooFormatter {
    fun format(): String {
        return "foo"
    }
}

@Component
internal class FooServiceField {
    @Autowired
    private val fooFormatter: FooFormatter? = null
}

@Component
internal class FooServiceSetter {
    private var fooFormatter: FooFormatter? = null

    @Autowired
    fun setFormatter(fooFormatter: FooFormatter?) {
        this.fooFormatter = fooFormatter
    }
}

@Component
internal class FooServiceConstructor @Autowired constructor(private val fooFormatter: FooFormatter)


